File: src/CorporateMessengerWithAgile/Application/Dto/Summary/TeamSummaryDto.cs
namespace Application.Dto.Summary
{
    public class TeamSummaryDto : BaseDto
    {
        public Guid ProjectId { get; set; }
        public string Title { get; set; } = null!;
        public int StandardSprintDuration { get; set; }
        public IReadOnlyList<Guid> TeamMemberIds { get; set; } = null!;
        public IReadOnlyList<Guid> SprintIds { get; set; } = null!;
        public IReadOnlyList<Guid> KanbanBoardColumnIds { get; set; } = null!;
    }
}

File: src/CorporateMessengerWithAgile/ReactApp/reactapp.client/src/models/entity/TeamSummaryDto.ts
// src/models/entity/TeamSummaryDto.ts
import { z } from 'zod';
import { BaseDtoSchema } from './BaseDto';
import { GuidSchema } from '../Guid';

export const TeamSummaryDtoSchema = BaseDtoSchema.extend({
    projectId: GuidSchema,
    title: z.string().min(1),
    standardSprintDuration: z.number(),
    teamMemberIds: z.array(GuidSchema).optional(),
    sprintIds: z.array(GuidSchema).optional(),
    kanbanBoardColumnIds: z.array(GuidSchema).optional(),
});

export type TeamSummaryDto = z.infer<typeof TeamSummaryDtoSchema>;
